Apps/W1/External File Storage - Azure File Service Connector/app/src/ExtFileShareConnectorImpl.Codeunit.al

```diff
@@ -450,6 +453,18 @@ codeunit 4570 "Ext. File Share Connector Impl" implements "External File Storage
         exit('/');
     end;
 
+    local procedure CreateUri(AccountId: Guid; SourcePath: Text): Text
+    var
+        FileShareAccount: Record "Ext. File Share Account";
+        TypeHelper: Codeunit "Type Helper";
+        FileShareUriLbl: Label 'https://%1.file.core.windows.net/%2/%3', Locked = true;
+    begin
+        FileShareAccount.Get(AccountId);
+        FileShareAccount.TestField("Storage Account Name");
+        FileShareAccount.TestField("File Share Name");
+        exit(StrSubstNo(FileShareUriLbl, FileShareAccount."Storage Account Name", FileShareAccount."File Share Name", TypeHelper.UrlEncode(SourcePath)));
```

Comment:
The Uri Codeunits provides a similiar method `Uri.EscapeDataString`.
I'm unsure if `UrlEncode` or `EscapeDataString` is the preferrable method here.

Reply:
I know the Uri.EscapeDataString procedure, but we escape here a part of the Url and not data in a post or get variable.

Reply:
@pri-krise you are right Uri.EscapeDataString is the only way to use it with spaces in the filename.